src/pages/Login-Signup/login.jsx

```diff
@@ -0,0 +1,72 @@
+import "../../root.css";
+import "../Login-Signup/auth.css";
+import { useAuth } from "../../context/authContext.js";
+import { Link, useNavigate } from "react-router-dom";
+import { useState, useEffect } from "react";
+
+const Login = () => {
+const { token, loginHandler } = useAuth();
+const navigate = useNavigate();
+const [ userDetails, setUserDetails] = useState({email: "", password: ""});
+
+// const guestUserDetails = (e) =>{
+//     e.preventDefault();
+// setUserDetails({
+// email: "adarshbalika@gmail.com",
+// password: "adarshBalika123"
+// })
+// }
+
+const guestUserHandler = (event) => {
+    event.preventDefault();
+    setUserDetails({
+        email: "adarshbalika@gmail.com",
+        password: "adarshBalika123",
+    })
+}
+
+function formHandler(e){
+    e.preventDefault();
+      loginHandler(userDetails);
+}
+
+// useEffect(()=>{
+// if(token){
+// navigate("/homepage");
+// }
+// },[token,navigate]);
+
+return(
+<div className="login-container justify-align">
+    <form className="form-container" onSubmit={formHandler}>
+        <h2 className="login-head">Login</h2>
+        <fieldset>
+            <legend for="username" className="inp-txt">Email</legend>
+            <input type="text" id="username" name="username" placeholder="abc@gmail.com" className="input-box"
+                value={userDetails.email} onChange={(e) => setUserDetails({...userDetails, email: e.target.value})} required /><br />
```

Comment:
Try avoid using <br/> tags while developing components!!
One more thing is that the onChange for both the fields have similar functions itself so you can write a common function itself and pass it in the onChange itself that just takes the event itself and updates userDetails by taking NAME and target of that event.
You can ask queries if you have difficulties framing it up

Reply:
Okay